balancer-js/src/modules/data/liquidity-gauges/provider.ts

```diff
@@ -69,15 +71,40 @@ export class LiquidityGaugeSubgraphRPCProvider
         gaugeAddresses
       );
     }
-    // TODO: Switch to getting rewards tokens from subgraph when indexer on polygon is fixed
-    // const rewardTokens = gauges.reduce((r: { [key: string]: string[] }, g) => {
-    //   r[g.id] ||= g.tokens ? g.tokens.map((t) => t.id.split('-')[0]) : [];
-    //   return r;
-    // }, {});
-    this.rewardData = await this.multicall.getRewardData(
-      gaugeAddresses //,
-      // rewardTokens
+
+    // Kept as a potential fallback for getting rewardData from RPC
+    // this.rewardData = await this.multicall.getRewardData(
+    //   gaugeAddresses //,
+    //   // rewardTokens
+    // );
+
+    // Reward data was made available from subgraph, keeping it separate for potential RPC fallback
+    this.rewardData = gauges.reduce(
+      (r: { [key: string]: { [key: string]: RewardData } }, g) => {
```

Comment:
How about we make these variable names more descriptive? I know it's trivial to identify that r means rewardData, g means gauge and t means token, but it would be way easier to just read the variables instead.
Besides that, I'm not sure if it's just me, but I find it really hard to read reducers like this. I'd love to see an alternative implementation to see if it's the logic that is indeed complex or if is just the reducer that makes it more complicated to understand.
My feeling is that we should optimize for readability. For such small pieces of code, I feel like it would take a few seconds to go through and review what's being done and instead I spend more than 5 min to make sense of what's going on 😅

Reply:
I agree with Bruno on the goal of aiming for readability. In the interests of time I'm going to merge this PR as is so we can release the fix in the latest package.